react-leaflet/index.d.ts

```diff
@@ -0,0 +1,258 @@
+// Type definitions for react-leaflet 1.0
+// Project: https://github.com/PaulLeCam/react-leaflet
+// Definitions by: Dave Leaver <https://github.com/danzel>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+//tslint:disable:no-empty-interface
+
+/// <reference types="geojson" />
+
+import * as Leaflet from 'leaflet';
+import * as React from 'react';
+
+//All events need to be lowercase so they don't collide with React.DOMAttributes<T>
```

Comment:
react-leaflet applies `toLowerCase` to all of the `on...` methods before using them for the internal leaflet library, so this works correctly. Both `onclick` and `onCLiCk` would work.
I've manually verified this as well.

https://github.com/PaulLeCam/react-leaflet/blob/d7ba0947b8fda95c1cdc54f1c5022cfd38306647/src/MapComponent.js#L42